File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnResponseMatches.java

package com.gracenote.gnsdk;

/** 
*  Collection of matches from a Scan & Match query 
*/ 
 
public class GnResponseMatches extends GnDataObject {
  private transient long swigCPtr;

  protected GnResponseMatches(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnResponseMatches_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnResponseMatches obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnResponseMatches(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

/** 
* Iterator for accessing response match(es) 
* @return Iterator 
*/ 
 
  public GnMatchIterable matches() {
    return new GnMatchIterable(gnsdk_javaJNI.GnResponseMatches_matches(swigCPtr, this), true);
  }

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnResponseMatches.cs

namespace GracenoteSDK {

/**
*  Collection of matches from a Scan & Match query
*/
public class GnResponseMatches : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnResponseMatches(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnResponseMatches_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnResponseMatches obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnResponseMatches() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnResponseMatches(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnMatchEnumerable Matches {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnResponseMatches_Matches_get(swigCPtr);
      GnMatchEnumerable ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnMatchEnumerable(cPtr, true);
      return ret;
    } 
  }

}

}
